HeterogeneousCore/CUDAUtilities/interface/cudaCompat.h

```diff
@@ -39,6 +39,11 @@ namespace cms {
       return old;
     }
 
+    template <typename T1, typename T2>
+    T1 atomicCAS_block(T1* address, T1 compare, T2 val) {
+      return atomicCAS(address, compare, val);
```

Comment:
for my education,
what does this and the other `_block` methods do in this PR, especially in the context of speeding things up.